src/lib/debounce.ts

```diff
@@ -29,7 +29,7 @@ export const debounceByKey = <T extends (...args: any[]) => any>(fn: T, ms: numb
           await fn(...args)
           promise.resolve()
         } catch (e) {
-          promise.reject(e.message)
+          promise.reject((e as any).message)
```

Comment:
This is making me question types lol. Promise.reject (from futures) says that it takes an Error not a string. Not sure if we should change this line to:
`promise.reject((e as Error))`

@cazala @menduz halp

Reply:
you should do `promise.reject(e)`, `e.message` is string, `e is Error`

Reply:
Changed it to use `promise.reject(e)`